x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/ScriptFieldMapper.java

```diff
@@ -56,81 +73,118 @@ protected String contentType() {
         return CONTENT_TYPE;
     }
 
-    public static class Builder extends FieldMapper.Builder<Builder> {
-
+    public static class Builder extends ParametrizedFieldMapper.Builder {
+
+        private final Parameter<Map<String, String>> meta = new Parameter<>(
+            "meta",
+            true,
+            Collections.emptyMap(),
+            TypeParsers::parseMeta,
+            m -> m.fieldType().meta()
+        );
+        private final Parameter<String> runtimeType = Parameter.stringParam(
+            "runtime_type",
+            true,
+            mapper -> toType(mapper).runtimeType,
+            null
+        );
+        // TODO script and runtime_type can be updated: what happens to the currently running queries when they get updated?
+        // do all the shards get a consistent view?
+        private final Parameter<Script> script = new Parameter<>("script", true, null,
+            Builder::parseScript, mapper -> toType(mapper).script);
         private final ScriptService scriptService;
 
-        private String runtimeType;
-        private Script script;
-
         protected Builder(String name, ScriptService scriptService) {
-            super(name, FIELD_TYPE);
+            super(name);
             this.scriptService = scriptService;
         }
 
-        public void runtimeType(String runtimeType) {
-            this.runtimeType = runtimeType;
-        }
-
-        public void script(Script script) {
-            this.script = script;
+        @Override
+        protected List<Parameter<?>> getParameters() {
+            return List.of(meta, runtimeType, script);
         }
 
         @Override
         public ScriptFieldMapper build(BuilderContext context) {
-            if (runtimeType == null) {
-                throw new IllegalArgumentException("runtime_type must be specified");
+            if (runtimeType.getValue() == null) {
+                throw new IllegalArgumentException("runtime_type must be specified for script field [" + name + "]");
```

Comment:
Once #59291 is merged then you can set validators directly on the parameter

Reply:
nice I was surpised that there wasn't a way to do that.